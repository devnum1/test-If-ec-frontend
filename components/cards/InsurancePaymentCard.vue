// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="insurance-payment-card">
    <div class="insurance-head">
      <div class="insurance-head__inner">
        {{ $t('common.insurance_payment_amount') }}
      </div>
    </div>
    <div class="insurance-body">
      <div class="insurance-body__inner">
        <div class="insurance-body__inner-input">
          <input
            v-model="insuranceValue"
            name="insuranceValue"
            class=""
            value=""
            placeholder="---"
            @keyup="updatePrice()"
          />
          <div>
            {{ $t('common.yen') }}
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
<script type="text/javascript">
export default {
  data() {
    return {
      progress: 1,
      insuranceValue: null,
    };
  },
  mounted() {},
  methods: {
    formatPrice(value) {
      let val = (value / 1).toFixed(0).replace('.', ',');

      return val.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
    },
    updatePrice() {
      this.insuranceValue = this.insuranceValue.replace(/\D/g, '');
      this.insuranceValue = this.formatPrice(parseInt(this.insuranceValue));
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/components/cards';
</style>
