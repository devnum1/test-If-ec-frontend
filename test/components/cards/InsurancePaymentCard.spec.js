import { shallowMount, config } from '@vue/test-utils';
import InsurancePaymentCard from '@/components/cards/InsurancePaymentCard.vue';
import flushPromises from 'flush-promises';

config.mocks.$t = (key) => key;

describe('InsurancePaymentCard', () => {
  let wrapper;

  const factory = (computed = {}) => {
    return shallowMount(InsurancePaymentCard);
  };

  beforeEach(() => {
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Card input keyup to call function formated price', async () => {
    wrapper = await factory();
    wrapper.setData({
      insuranceValue: '3725600',
    });

    await flushPromises();

    const input = wrapper.find('input');

    input.trigger('keyup');
    await flushPromises();
    expect(wrapper.vm.insuranceValue).toEqual('3,725,600');
  });
});
