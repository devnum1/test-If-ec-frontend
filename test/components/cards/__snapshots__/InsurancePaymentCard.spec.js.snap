// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsurancePaymentCard match Snapshot 1`] = `
"<div class=\\"insurance-payment-card\\">
  <div class=\\"insurance-head\\">
    <div class=\\"insurance-head__inner\\">
      common.insurance_payment_amount
    </div>
  </div>
  <div class=\\"insurance-body\\">
    <div class=\\"insurance-body__inner\\">
      <div class=\\"insurance-body__inner-input\\"><input name=\\"insuranceValue\\" value=\\"\\" placeholder=\\"---\\">
        <div>
          common.yen
        </div>
      </div>
    </div>
  </div>
</div>"
`;
